controller/api/public/proto_over_http.go

```diff
@@ -118,32 +118,18 @@ func serializeAsPayload(messageContentsInBytes []byte) ([]byte, error) {
 }
 
 func deserializePayloadFromReader(reader *bufio.Reader) ([]byte, error) {
-	messageLengthInBytes := make([]byte, numBytesForMessageLength)
-	reader.Read(messageLengthInBytes)
-	messageLength := int(binary.LittleEndian.Uint32(messageLengthInBytes))
-
-	messageContentsAsBytes := make([]byte, 0)
-	totalBytesRead := 0
-	for {
-		remainingBytes := messageLength - totalBytesRead
-		buffer := make([]byte, remainingBytes)
-		numBytesRead, err := reader.Read(buffer)
-		messageRead := buffer[:numBytesRead]
-		messageContentsAsBytes = append(messageContentsAsBytes, messageRead...)
-		if err != nil {
-			if err == io.EOF {
-				if totalBytesRead != messageLength {
-					return nil, fmt.Errorf("message declared size [%d] but could only read [%d] bytes before an EOF", messageLength, totalBytesRead)
-				}
-				break
-			}
-			return nil, err
-		}
-		totalBytesRead = totalBytesRead + numBytesRead
+	messageLengthAsBytes := make([]byte, numBytesForMessageLength)
+	reader.Read(messageLengthAsBytes)
+	messageLength := int(binary.LittleEndian.Uint32(messageLengthAsBytes))
 
-		if totalBytesRead == messageLength {
-			break
-		}
+	messageContentsAsBytes := make([]byte, messageLength)
+	totalBytesRead, err := io.ReadFull(reader, messageContentsAsBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	if totalBytesRead != messageLength {
+		return nil, fmt.Errorf("message declared length [%d] but could only read [%d] bytes")
```

Comment:
Just to wrap my head around this error case, What scenarios would cause the code to reach this line when deserializing the HTTP? Proto? body.